numpy/lib/function_base.py

```diff
@@ -1621,6 +1621,56 @@ def trim_zeros(filt, trim='fb'):
     >>> np.trim_zeros([0, 1, 2, 0])
     [1, 2]
 
+    """
+    try:
+        return _trim_zeros_new(filt, trim)
+    except Exception as ex:
+        warning = DeprecationWarning(
+            "in the future trim_zeros will require a 1-D array as input"
+            "which is compatible with ndarray.astype(bool)"
```

Comment:
I don’t want to be a smart ass, but this is important.

The sentence should use “that” rather than “which”. The difference is that with “which” the sentence says that all 1-D arrays are compatible with astype(bool), but I guess you meant to say that the array must be 1-D AND compatible with astype(bool).

It’s a common mistake made by German natives, and probably also Dutch (from your name I assume you are Dutch).

Reply:
And English natives!  I often make that mistake myself...

Reply:
> The sentence should use “that” rather than “which”. The difference is that with “which” the sentence says that all 1-D arrays are compatible with astype(bool), but I guess you meant to say that the array must be 1-D AND compatible with astype(bool).

Out of curiosity, isn't there a distinction between cases where "which" is preceded by a comma ("... a 1-D array as input, which is compatible ...") versus ones without comma ("... a 1-D array as input which is compatible ...")? 
The first scenario definitely conveys the incorrect meaning, but I was under the impression that whether or not the second one is acceptable is more of a American vs British thing.

In any case, it should be changed for the sake of clarity (see https://github.com/numpy/numpy/pull/16911/commits/4ddfc3eb68202a626d8bb318963679f8e831b1e6).

Reply:
I *guess* there is no distinction because English is "insensitive" to commas in many cases (and also British and American place commas differently). But that's just a guess and I'd be interested in a definitive answer if anybody knows it :-)

Reply:
I think you're right, that the 'which' there makes it a little more likely that someone could get the impression that any 1D array can be cast as bool.  I agree that the impression is less using 'that'.   But like you, I couldn't give you a definitive source for that (or which, oh no, so confusing).